src/ui/EditorNS/editor.cpp

```diff
@@ -268,117 +305,184 @@ namespace EditorNS
         sendMessage("C_CMD_MARK_DIRTY");
     }
 
-    QList<QMap<QString, QString>> Editor::languages()
+    void Editor::initLanguageCache()
     {
-        // TODO: Cache this data.
-        QJSEngine engine;
         QFileInfo fileInfo(Notepadqq::editorPath());
-        QString fileName = fileInfo.absolutePath() + "/classes/Languages.js";
+        QString fileName = fileInfo.absolutePath() + "/classes/Languages.json";
         QFile scriptFile(fileName);
-        scriptFile.open(QIODevice::ReadOnly);
-        QTextStream stream(&scriptFile);
-        QString contents = stream.readAll();
+        scriptFile.open(QIODevice::ReadOnly | QIODevice::Text);
+        QJsonDocument json = QJsonDocument::fromJson(scriptFile.readAll());
         scriptFile.close();
-
-        QJSValue result = engine.evaluate(contents, fileName);
-
-        if (result.isError()) {
-            qDebug() << "Failed to load languages file.";
-            return QList<QMap<QString, QString>>();
+        // Begin iterating through our language objects.
+        for (auto it = json.object().constBegin(); it != json.object().constEnd(); ++it) {
+            auto mode = it.value().toObject();
+            LanguageData newMode;
+            newMode.id = it.key();
+            newMode.name = mode.value("name").toString();
+            newMode.mime = mode.value("mime").toString();
+            newMode.mode = mode.value("mode").toString();
+            newMode.fileNames = mode.value("fileNames").toVariant().toStringList();
+            newMode.fileExtensions = mode.value("fileExtensions").toVariant().toStringList();
+            newMode.firstNonBlankLine = mode.value("firstNonBlankLine").toVariant().toStringList();
+
+            m_langCache.append(newMode);
         }
+    }
 
-        QMap<QString, QVariant> languages =
-            engine.evaluate("Languages.languages", fileName).toVariant().toMap();
-
-        QList<QMap<QString, QString>> langs;
-
-        QMap<QString, QVariant>::iterator lang;
-        for (lang = languages.begin(); lang != languages.end(); ++lang) {
-            QMap<QString, QVariant> mode = lang.value().toMap();
-
-            QMap<QString, QString> newMode;
-            newMode.insert("id", lang.key());
-            newMode.insert("name", mode.value("name").toString());
-            newMode.insert("mime", mode.value("mime").toString());
-            newMode.insert("mode", mode.value("mode").toString());
-            langs.append(newMode);
+    QVariant Editor::getLanguageVariantData(const QString& langId)
+    {
```

Comment:
It may be unused now that I have the LanguageData struct... I'll have to double check to verify before removing.